File: src/spirite/base/pen/behaviors/ZoomingReferenceBehavior.java
package spirite.base.pen.behaviors;

import spirite.base.pen.Penner;

public class ZoomingReferenceBehavior extends StateBehavior {
	public ZoomingReferenceBehavior(Penner penner) {
		super(penner);
	}
	int startx = this.penner.x;
	int starty = this.penner.y;
	@Override public void start() {}
	@Override public void onTock() {}
	@Override
	public void onMove() {
		this.penner.workspace.getReferenceManager().zoomTransform(
				(float)Math.pow(1.0015, 1+(this.penner.rawY - this.penner.oldRawY)), startx, starty);
	}
}

File: src/main/kotlin/spirite/base/pen/behaviors/ZoomingReferenceBehavior.kt
package spirite.base.pen.behaviors

import rb.vectrix.mathUtil.d
import rb.vectrix.mathUtil.f
import spirite.base.pen.Penner

class ZoomingReferenceBehavior(penner: Penner) : PennerBehavior(penner) {

    private val startX = penner.x
    private val startY = penner.y

    override fun onStart() {}
    override fun onTock() {}
    override fun onMove() {
        penner.workspace?.referenceManager?.zoomTransform(
                Math.pow(1.0015, 1 + (penner.rawY - penner.oldRawY).d).f, startX, startY) ?: end()
    }

}